### ISSUE TITLE
Content of the "Media Replace Button" are shifted sideways

### ISSUE THREAD

**Issue by ddryo** [WordPress/gutenberg#27186]
Thank you for your continuous development.

I'm sorry if an issue has already been raised.
While testing version 5.6RC, I found something like a bug, so I'll report it.

When you click the replace button of the image that can be specified from the toolbar in "Image block" etc., the contents are shifted sideways.

The following image is an image of the bug reproduced on Twenty Twenty One.

<img width="713" alt="image-replace-btn-56" src="https://user-images.githubusercontent.com/31400297/99937800-df0fa700-2da9-11eb-9b84-0ca42967c39e.png">

**My testing environment:**
 - OS: macOS Catalina 10.15.7
 - Browser: Chrome
 - WordPress: 5.6 RC1
 - Language: Japanese
 - Theme: Twenty Twenty One

**Comment by JustinyAhin** [comment]
I am able to reproduce this. In ja locale but also in others.
Worth noting that it only appears on Chrome.

**Comment by michaelbourne** [comment]
While a CSS edit of `overflow-y: inherit;` on the parent does fix this, the root cause is deeper. The screenreader text beside the media link URL for example is set to `position: absolute;`. Setting the parent `.block-editor-media-flow__url-input .block-editor-link-control .block-editor-link-control__search-item-title` to `position: relative;` fixes the overflow, but the external link icon is still hidden. 

A better solution would be to wrap the URL itself in another span and set the overflow and ellipsis on that element, with a max width narrow enough to show the external link icon (roughly 160px). (And then remove the overflow and ellipsis on `.block-editor-link-control__search-item .block-editor-link-control__search-item-title`)

```
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 160px;
    display: inline-block;
    line-height: 1;
```

With the URL fragment wrapped in it's own `span`, the above styles fix both the overflow as well as the hidden external icon SVG.

**Comment by ddryo** [comment]
I confirmed that it still reproduces in 5.6.

After investigating the cause, it seems that `components-external-link__icon` is popping out by the length of the "URL of the image".

<img width="628" alt="56_bug_2020-12-09 9 42 58" src="https://user-images.githubusercontent.com/31400297/101559033-8c69f800-3a03-11eb-8465-34813926d674.png">

So, I also found that if the image URL is long, the screen will be blank as shown below.

<img width="622" alt="56_bug_2020-12-09 9 31 00" src="https://user-images.githubusercontent.com/31400297/101559084-ae637a80-3a03-11eb-94a1-a26db016d16a.png">

I think it is necessary to enclose the "URL part" with tags such as `<span>`.

**Comment by ddryo** [comment]
As a temporary workaround, the following CSS has been added to prevent scrolling from occurring.

```css
.block-editor-media-flow__url-input {
    overflow: hidden;
}
```

<img width="474" alt="56_bug_2020-12-09 9 51 38" src="https://user-images.githubusercontent.com/31400297/101559555-b2dc6300-3a04-11eb-867d-f58931d46720.png">

However, with this, the SVG icon that is popping out remains invisible.

**Comment by simison** [comment]
Same as https://github.com/WordPress/gutenberg/issues/27653

**Comment by jg314** [comment]
For those looking for a temporary fix, we added this CSS to load within the block editor:

```css
.block-editor-link-control__search-item .block-editor-link-control__search-item-title {
  position: relative;
}
```

It's not a perfect solution, but it does fix the biggest usability issue, the horizontal scrollbar.

**Comment by michaelbourne** [comment]
Further testing on my end reaffirms my findings above. There needs to be a small DOM edit before any CSS changes are made in order to properly fix this.

**Comment by JustinyAhin** [comment]
@michaelbourne can you please test again with the latest version if Gutenberg activated? It looks like this has been fixed in a recent version of the plugin.

**Comment by speedyprem** [comment]
@talldan I have created a PR for this issue: https://github.com/WordPress/gutenberg/pull/28604

**Comment by paaljoachim** [comment]
I noticed this is also an issue in WP 5.6.1.

It is not an issue in WP 5.7 beta 2. Meaning a fix has been made somewhere between 5.6 and 5.7.

**Comment by draganescu** [comment]
Howdy friends 👋  This seems to have been solved in #27777 so I will close the issue. Thank you for reporting and for all the helpful comments.
